contracts/SolverTrampoline.sol

```diff
@@ -53,6 +56,10 @@ contract SolverTrampoline {
         }
         nonces[solver] = nonce + 1;
 
+        if (block.number > deadline) {
+            revert Expired();
```

Comment:
Publishing the current and deadline block number may make debugging slightly easier.

Reply:
Should I do the same for the `Nonce` error as well?

Reply:
As you feel (but I think it could be a good idea). I'm just trying to think ahead what could make debugging this easier at a later date.

Reply:
In general I agree with your suggestion. It was more of "do you think its worth a follow-up PR to add `InvalidNonce` error context".

I will do it. Its easy PR anyway.